AlternativeChess/ChessBoardViewController.swift

```diff
@@ -16,24 +24,58 @@ struct Square {
         case black
     }
     let color: Color
+    var sideOfSquare = 0.0
 }
 
 class ChessBoardView: UIView {
-    private static let squaresInRow = 8
+    static let squaresInRow = 8
+
+    override init(frame: CGRect) {
+         super.init(frame: frame)
 
-    let squares: [[Square]] = {
+     }
+
+     required init?(coder: NSCoder) {
+         super.init(coder: coder)
+     }
+    
+    var squares: [[Square]] = {
         var board: [[Square]] = []
         for row in 0 ..< ChessBoardView.squaresInRow {
             board.append([])
             for column in 0 ..< ChessBoardView.squaresInRow {
-                let squareColor: Square.Color = (row + column).isMultiple(of: 2) ? .black : .white
+                let squareColor: Square.Color = (row + column).isMultiple(of: 2) ? .white : .black
                 board[row].append(Square(color: squareColor))
             }
         }
         return board
     }()
 
     override func draw(_ rect: CGRect) {
-        // TODO: draw CGRect based on square in rect
+        let side = bounds.width / CGFloat(ChessBoardView.squaresInRow)
+        var row = 0
+        for(_, item) in squares.enumerated() {
+            var counter = 0
+            for square in item {
+                 if (square.color == .white) {
+                     let path = UIBezierPath(rect: CGRect(x: CGFloat(counter) * side, y: CGFloat(row) * side, width: side, height: side))
```

Comment:
We can simplify this code. the path variable can be created outside the if statement since in both cases it is initialized to the same value